debian/matrix-synapse-py3.postinst

```diff
@@ -43,12 +44,19 @@ EOF
       adduser --quiet --system --group --no-create-home --home /var/lib/matrix-synapse $USER
     fi
 
-    for DIR in /var/lib/matrix-synapse /var/log/matrix-synapse /etc/matrix-synapse; do
+    for DIR in /var/log/matrix-synapse; do
       if ! dpkg-statoverride --list --quiet $DIR >/dev/null; then
         dpkg-statoverride --force-statoverride-add --quiet --update --add $USER "$(id -gn $USER)" 0755 $DIR
       fi
     done
 
+    if ! dpkg-statoverride --list --quiet /var/lib/matrix-synapse /etc/matrix-synapse >/dev/null; then
+      dpkg-statoverride --force-statoverride-add --quiet --update --add $USER "$(id -gn $USER)" 0770 /var/lib/matrix-synapse
+    fi
+
+    if ! dpkg-statoverride --list --quiet $CONFIGFILE_HOMESERVER >/dev/null; then
+      dpkg-statoverride --force-statoverride-add --quiet --update --add $USER "$(id -gn $USER)" 0660 $CONFIGFILE_HOMESERVER
+    fi
```

Comment:
is this still necessary? It feels a bit redundant (and possibly confusing, given other files in /etc/matrix-synapse will be unchanged)

Reply:
In my opinion it would be better to explicitly restrict permissions on _both_ the directory and configuration file. However the result is the same, and I could not find any other example package that would indicate what the best practice is. The closest I could find on my systems is the [transmission-daemon](https://packages.debian.org/bullseye/transmission-daemon) package, for which the configuration directory is world-readable, but the configuration file itself (contains a password) is not:

```
srvadm@demo1:~$ ls -ld /etc/transmission-daemon/
drwxrwsr-x 2 root debian-transmission 4096 Apr  7 13:53 /etc/transmission-daemon/
srvadm@demo1:~$ ls -l /etc/transmission-daemon/settings.json 
-rw------- 1 debian-transmission debian-transmission 2384 Apr  7 13:53 /etc/transmission-daemon/settings.json
```

So I guess only having `/etc/matrix-synapse` as `0770` and leaving the file itself untouched is OK. What do you think?

Reply:
> As there already is an existing `dpkg-statoverride` on `/etc/matrix-synapse`, it will not be updated and the directory will stay mode `0755`.

Good spot. But it sounds to me like this is the problem: we should be fixing the existing statoverride, not working around it.

One option for that is to add a command which removes the broken statoverride on upgrade:

```shell
    for DIR in /var/lib/matrix-synapse /etc/matrix-synapse; do
      # if we are upgrading from synapse <= 1.81.0, and a stat-override exists but has the wrong permissions,
      # that is a result of https://github.com/matrix-org/synapse/issues/6364, and we should fix it.
      if dpkg --compare-versions "$2" le-nl "1.81.0" && 
          [ "$(dpkg-statoverride --list $DIR)" = "$USER $(id -gn $USER) 0755 $DIR"]; then
        echo "Fixing incorrect permissions on $DIR due to https://github.com/matrix-org/synapse/issues/6364"
        dpkg-statoverride --remove $DIR
      fi

      if !dpkg-statoverride --list $DIR; then
        dpkg-statoverride --update --add $USER "$(id -gn $USER)" 0770 $DIR
      fi
```

Reply:
also, let's pull out a `$GROUP` variable rather than writing `$(id -gn $USER)` repeatedly?